spec/models/page_spec.rb

```diff
@@ -256,5 +263,62 @@
       expect(Page.new.action_count).to eq 0
     end
   end
+
+  describe 'friendly_id' do
+    let!(:page) { create(:page, title: 'simple slug') }
+
+    it 'generates slug' do
+      expect(page.friendly_id).to eq('simple-slug')
+      expect(page.slug).to        eq('simple-slug')
+    end
+
+    context 'finder' do
+      it 'finds by slug' do
+        expect(Page.find('simple-slug')).to eq page
+      end
+
+      it 'finds by id' do
+        expect(Page.find(page.id)).to eq page
+      end
+
+      it 'finds using friendly.find' do
+        expect(Page.friendly.find('simple-slug')).to eq page
+        expect(Page.friendly.find(page.id)).to eq page
+      end
+    end
+
+    context 'updating title' do
+      before do
+        page.update title: 'Complex Slug', slug: nil
```

Comment:
According to the docs, `slug` must be nil for friendly_id to generate a new one.